logstash-core/src/main/java/org/logstash/ackedqueue/io/FileCheckpointIO.java

```diff
@@ -64,10 +63,12 @@ public Checkpoint write(String fileName, int pageNum, int firstUnackedPageNum, l
 
     @Override
     public void write(String fileName, Checkpoint checkpoint) throws IOException {
-        Path path = Paths.get(dirPath, fileName);
-        final byte[] buffer = new byte[BUFFER_SIZE];
         write(checkpoint, buffer);
-        Files.write(path, buffer, WRITE_OPTIONS);
+        buffer.flip();
+        try (FileOutputStream out = new FileOutputStream(Paths.get(dirPath, fileName).toFile())) {
```

Comment:
This also changes the file writing strategy from Files.write to FileOutputStream, can you tell me about this change?